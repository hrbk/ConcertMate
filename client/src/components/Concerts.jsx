import React from 'react';
import axios from 'axios';
import ConcertEntry from './ConcertEntry.jsx';
import {ListGroup} from 'react-bootstrap';
import {Panel, Tabs, Tab} from 'react-bootstrap';
import { connect } from 'react-redux';
import ReactScrollbar from 'react-scrollbar-js';

const Concerts = (props) => {
  const scroll = {
    width: '100%',
    height: '400',
    speed: '10'
  }
  const nearby = (
    <ReactScrollbar style={scroll}>
      <Panel collapsible defaultExpanded header={`Nearby Concerts`}>
        <ListGroup fill>
          {props.events.map((event, i) => {
            return <ConcertEntry event={event} key={i} handleArtistClick={props.handleArtistClick} />
          })}
        </ListGroup>
      </Panel>
    </ReactScrollbar>
  )
  const popularityObj = (events) => {
    props.events.sort((a, b) => {
      if (a.popularity > b.popularity) {
        return 1;
      }
      if (a.popularity < b.popularity) {
        return -1;
      }
      return 0;
    }).reverse();
  }

  const popular = (
    <ReactScrollbar style={scroll}>
      <Panel collapsible defaultExpanded header={`Popular Concerts`}>
        <ListGroup fill>
          {popularityObj(props.events)}
          {props.events.map((event, i) => {
            return <ConcertEntry event={event} key={i} handleArtistClick={props.handleArtistClick} />
          })}
        </ListGroup>
      </Panel>
    </ReactScrollbar>
  )

  return (
    <div>
      <Tabs defaultActiveKey={1}>
        <Tab eventKey={1} title='Nearby'>{nearby}</Tab>
        <Tab eventKey={2} title='Popular'>{popular}</Tab>
      </Tabs>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    'events': state.events
  }
}

export const UnwrappedConcerts = Concerts;
export default connect(mapStateToProps)(Concerts);
