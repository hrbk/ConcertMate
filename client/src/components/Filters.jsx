import React from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import {Navbar, FormGroup, FormControl, Button} from 'react-bootstrap';
import PlacesAutocomplete, { geocodeByAddress, getLatLng } from 'react-places-autocomplete';
import axios from 'axios';
import { setDate, setSearchTerm, setMapCenter } from '../redux/actionCreators.js';
import { connect } from 'react-redux';


// ignore the fact that this is called Favorites but the file is called Filters
const Filters = (props) => {
  // beginning of search functionality. we wanted to implement google search to be able to
  // autocomplete addresses but hey that's your job now
  const handleSubmit = (loc) => {

    geocodeByAddress(loc)
    .then(results => getLatLng(results[0]))
    .then(latLng => {

      let userLoc = {
        lat: latLng.lat,
        lng: latLng.lng
      }
      props.handleSetMapCenter(userLoc); //set state of mapCenter prop
      props.requestEvents(); //getSongkickEvents

    })
    .catch(error => console.log('error', error))
  }

  const datepicker =  {
    paddingTop: '3.5px'
  }

  const inputProps = {
      value: props.searchTerm,
      onChange: props.handleSetSearchTerm,
      class: 'search_input',
      placeholder: 'Location'
  }

  const navStyle = {
    display: 'flex',
    alignItems: 'center'
  }

  return (
    <div className="navigation">
      <Navbar.Form pullRight style={navStyle}>
        <PlacesAutocomplete
          onEnterKeyDown={handleSubmit}
          inputProps={inputProps}
          highlightFirstSuggestion={true}
          googleLogo={false}/>

        <div style={datepicker}>
          <DatePicker
            dateFormat="MM/DD/YYYY"
            selected={props.date} //automatically passed into handleSetData
            onChange={props.handleSetDate} />
        </div>
        <Button type="submit" onClick={() => handleSubmit(props.searchTerm)}>Submit</Button>
      </Navbar.Form>
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    'date': state.date, //allows this.props.date to exist and be accessible in component
    'searchTerm': state.searchTerm,
    'mapCenter': state.mapCenter
  }
}

const mapDispatchToProps = (dispatch) => ({
  handleSetDate(date = moment()) { //think of this as used in place of 'this.setState()'
    dispatch(setDate(date)); //action creator dispatched and reducers called: { type: 'SET_SPOTIFY_TOKEN', value: token } equivalent
  },
  handleSetSearchTerm(searchTerm = '') {
    dispatch(setSearchTerm(searchTerm)); //no need for searchTerm.target.value bc googleAutoPlaces handles that
  },
  handleSetMapCenter(mapCenter) {
    dispatch(setMapCenter(mapCenter));
  }
});

export const UnwrapperFilters = Filters;
export default connect(mapStateToProps, mapDispatchToProps)(Filters);
