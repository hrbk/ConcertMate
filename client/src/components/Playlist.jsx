import React from 'react';
import axios from 'axios';
import {Panel} from 'react-bootstrap'
import { connect } from 'react-redux';

const Playlist = (props) => {
	let iframeBaseURL = 'https://open.spotify.com/embed?uri=spotify:';
	let iframeQuery = props.artistId ? 'artist:' + props.artistId + '&theme=black' : 'user:1211115253:playlist:6r1hzKf8a0KYkbt44we1Cl';
	let iframeSRC = iframeBaseURL + iframeQuery;

  return (
  	<div>
	    <Panel collapsible defaultExpanded header="Preview Artist" bsStyle="success">
	    	<iframe src={iframeSRC}
					width="100%" height="95"
					frameBorder="0" allowTransparency="true"></iframe>

	    </Panel>
    </div>
  )
};


const mapStateToProps = (state) => {
  return {
    'artistId': state.artistId //allows this.props.artistId to exist and be accessible in component
  }
}

export const UnwrappedPlaylist = Playlist;
export default connect(mapStateToProps)(Playlist);
