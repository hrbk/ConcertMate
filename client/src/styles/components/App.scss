.concert-mate-app {
  #map {
    height: 100%;
    width: 100%;
  }
  .map-container {
    height: 92.6vh;
    width: 100%;
  }
  .navigation > .navbar {
    margin-bottom: 0;
  }
  .navbar-brand small {
    margin-left: 5px;
  }
  > .container > .row {
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
input {
  font-size: 12px;
}
#PlacesAutocomplete__root {
  margin: 5px;
}
.playlist-container {
  position: fixed;
  width: 360px;
  bottom: 0;
  right: 60px;

  .panel {
    border: none;
    margin-bottom: 0;
  }
  .panel-success {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
  }
  .panel-heading {
    border-top-left-radius: 0;
    border-top-right-radius: 0;
  }
  .panel-success > .panel-heading {
    border: none;
  }
}

@media screen and (max-width: 767px) {
  .playlist-container {
    height: 300px;
  }
  .concert-mate-app .map-container {
    height: 95vh;
    margin: 0;
  }
}

@media screen and (max-width: 650px) {
  .concert-mate-app > .container > .row {
    flex-flow: column;
  }
}

@media screen and (max-width: 420px) {
  .playlist-container {
    width: 70%;
  }
}
