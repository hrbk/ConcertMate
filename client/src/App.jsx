import React from 'react';
import { Grid, Row, Col } from 'react-bootstrap';
import axios from 'axios';
import moment from 'moment';
import Filters from './components/Filters.jsx';
import Map from './components/Map.jsx';
import Playlist from './components/Playlist.jsx';
import Concerts from './components/Concerts.jsx';
import ReactScrollbar from 'react-scrollbar-js';
import { PageHeader, Navbar } from 'react-bootstrap';
import { connect } from 'react-redux';
import { setSpotifyToken, setArtist, setSongkickEvents, setArtistId } from './redux/actionCreators.js';
import './styles/main.scss';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.requestSongkickEvents = this.requestSongkickEvents.bind(this);
    this.handleArtistClick = this.handleArtistClick.bind(this);
  }

  componentDidMount() {
    this.authenticateSpotify();
    this.requestSongkickEvents();
  }

  authenticateSpotify() {
    if (window.location.hash) {
      let hash = window.location.hash;
      let token = hash.split('&')[0].split('=')[1];
      this.props.handleSetSpotifyToken(token);

      axios.post('/spotify/login', {
          data: token
      })
      .catch((error) => {
        console.log(error);
      });

    } else {
      axios.get('/spotify/login')
        .then((response) => {
          let loginUrl = response.data;
          window.location = loginUrl;
        })
        .catch((error) => {
          console.log(error);
        });
    }
  }

  handleArtistClick(clickedArtist) {
    this.props.handleSetArtist(clickedArtist);
    this.requestArtistId(clickedArtist);
  }

  requestArtistId(clickedArtist = this.props.artist) {
    if (clickedArtist) {
      let data = {
          artist: clickedArtist,
          token: this.props.token
        };

        axios.post('/spotify/search', data)
          .then((res) => {
            this.props.handleSetArtistId(res.data.artistId);
          })
          .catch((err) => {
            console.log(err);
          });

    } else {
      this.props.handleSetArtist();
    }
  }

  requestSongkickEvents() {
    let formattedDate = this.props.date.format('YYYY-MM-DD');
    let latitude = this.props.mapCenter.lat;
    let longitude = this.props.mapCenter.lng;
    console.log('COORDS', latitude, longitude);
    axios.post('/songkick/', {
      date: formattedDate,
      lat: latitude,
      lng: longitude,
      city: this.props.searchTerm
    })
      .then((data) => {
        this.props.handleSetSongkickEvents(data.data);
        this.props.handleSetArtist(data.data[0].headline);
        this.requestArtistId();
      })
      .catch((err) => {
        console.log('Error: ', err);
      });
  }

  popularityFilter() {
    console.log('EVENTS', this.props.events);
  }

 render() {
    const scrollbar = {
      width: '100%',
      height: '290',
    };

    const header = {
      fontFamily: 'futura',
      fontSize: 70,
      color: '#CD3F2A'
    };

    return (

      <div className="concert-mate-app">
        <div className="container">
          <Row>
            <Col md={6}>
              <Navbar.Header style={header}>
                <Navbar.Brand>
                  <a href="/">ConcertMate</a><small>discover upcoming concerts around you</small>
                </Navbar.Brand>
              </Navbar.Header>
            </Col>
            <Col md={6}>
              <Filters requestEvents={this.requestSongkickEvents} />
            </Col>
          </Row>
        </div>
        <Row className="map-container">
          <Map />
        </Row>
        <div className="playlist-container">
          <Playlist />
            <div id="concert-container">
              <Concerts events={this.props.events} handleArtistClick={this.handleArtistClick} popularityFilter={this.popularityFilter}/>
            </div>
        </div>
      </div>

    )
  }
}

const mapStateToProps = (state) => {
  return {
    'token': state.token, //allows this.props.token to exist and be accessible in component
    'date': state.date, //need access to date here but don't need to edit its state here, so no dispatch method associated
    'artist': state.artist,
    'events': state.events,
    'artistId': state.artistId,
    'mapCenter': state.mapCenter,
    'searchTerm': state.searchTerm
  }
}

const mapDispatchToProps = (dispatch) => ({
  handleSetSpotifyToken(token = '') { //think of this as used in place of 'this.setState()'
    dispatch(setSpotifyToken(token)); //action creator dispatched and root reducer called: { type: 'SET_SPOTIFY_TOKEN', value: token } equivalent
  },
  handleSetArtist(artist = '') {
    dispatch(setArtist(artist));
  },
  handleSetSongkickEvents(events = []) {
    dispatch(setSongkickEvents(events));
  },
  handleSetArtistId(artistId = '') {
    dispatch(setArtistId(artistId));
  }
});

export const UnwrappedApp = App;
export default connect(mapStateToProps, mapDispatchToProps)(App);
