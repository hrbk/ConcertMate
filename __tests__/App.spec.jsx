import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App,   { UnwrappedApp } from '../client/src/App.jsx';
import Filters from '../client/src/components/Filters.jsx';
import Map from '../client/src/components/Map.jsx';
import Playlist from '../client/src/components/Playlist.jsx';
import ReactScrollbar from 'react-scrollbar-js';
import Concerts from '../client/src/components/Concerts.jsx';
import { NavbarBrand } from 'react-bootstrap';
import { Provider } from 'react-redux';
import store from '../client/src/redux/store';

describe('<App />', () => {
  it('should render correctly', () => {
    const component = shallow(<UnwrappedApp />);
    expect(component).toMatchSnapshot();
  });

  it('should contain Filters, Map, Playlist, ReactScrollbar, Concerts', () => {
    const component = shallow(<UnwrappedApp />);
    expect(component
        .containsAnyMatchingElements([
        <Filters />,
        <Map />,
        <Playlist />,
        <ReactScrollbar />
      ])).toEqual(true);
  });

  it('should render brand with ConcertMate', () => {
    const component = shallow(<UnwrappedApp />);
    expect(component.find(NavbarBrand).find('a').text()).toEqual('ConcertMate');
  });

});
