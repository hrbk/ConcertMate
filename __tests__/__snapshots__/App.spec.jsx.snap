// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="concert-mate-app"
>
    <div
        className="container"
    >
        <Row
            bsClass="row"
            componentClass="div"
        >
            <Col
                bsClass="col"
                componentClass="div"
                md={6}
            >
                <NavbarHeader
                    style={
                        Object {
                            "color": "#CD3F2A",
                            "fontFamily": "futura",
                            "fontSize": 70,
                          }
                    }
                >
                    <NavbarBrand>
                        <a
                            href="/"
                        >
                            ConcertMate
                        </a>
                        <small>
                            discover upcoming concerts around you
                        </small>
                    </NavbarBrand>
                </NavbarHeader>
            </Col>
            <Col
                bsClass="col"
                componentClass="div"
                md={6}
            >
                <Connect(Filters)
                    requestEvents={[Function]}
                />
            </Col>
        </Row>
    </div>
    <Row
        bsClass="row"
        className="map-container"
        componentClass="div"
    >
        <Connect(Map) />
    </Row>
    <div
        className="playlist-container"
    >
        <Connect(Playlist) />
        <ScrollWrapper
            className="react-scrollbar-default"
            speed={53}
            style={
                Object {
                    "height": "290",
                    "width": "100%",
                  }
            }
        >
            <Connect(Concerts)
                events={undefined}
                handleArtistClick={[Function]}
            />
        </ScrollWrapper>
    </div>
</div>,
  "nodes": Array [
    <div
      className="concert-mate-app"
>
      <div
            className="container"
      >
            <Row
                  bsClass="row"
                  componentClass="div"
            >
                  <Col
                        bsClass="col"
                        componentClass="div"
                        md={6}
                  >
                        <NavbarHeader
                              style={
                                    Object {
                                          "color": "#CD3F2A",
                                          "fontFamily": "futura",
                                          "fontSize": 70,
                                        }
                              }
                        >
                              <NavbarBrand>
                                    <a
                                          href="/"
                                    >
                                          ConcertMate
                                    </a>
                                    <small>
                                          discover upcoming concerts around you
                                    </small>
                              </NavbarBrand>
                        </NavbarHeader>
                  </Col>
                  <Col
                        bsClass="col"
                        componentClass="div"
                        md={6}
                  >
                        <Connect(Filters)
                              requestEvents={[Function]}
                        />
                  </Col>
            </Row>
      </div>
      <Row
            bsClass="row"
            className="map-container"
            componentClass="div"
      >
            <Connect(Map) />
      </Row>
      <div
            className="playlist-container"
      >
            <Connect(Playlist) />
            <ScrollWrapper
                  className="react-scrollbar-default"
                  speed={53}
                  style={
                        Object {
                              "height": "290",
                              "width": "100%",
                            }
                  }
            >
                  <Connect(Concerts)
                        events={undefined}
                        handleArtistClick={[Function]}
                  />
            </ScrollWrapper>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleArtistClick": [Function],
        "props": Object {},
        "refs": Object {},
        "requestSongkickEvents": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="concert-mate-app"
>
          <div
                    className="container"
          >
                    <Row
                              bsClass="row"
                              componentClass="div"
                    >
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={6}
                              >
                                        <NavbarHeader
                                                  style={
                                                            Object {
                                                                      "color": "#CD3F2A",
                                                                      "fontFamily": "futura",
                                                                      "fontSize": 70,
                                                                    }
                                                  }
                                        >
                                                  <NavbarBrand>
                                                            <a
                                                                      href="/"
                                                            >
                                                                      ConcertMate
                                                            </a>
                                                            <small>
                                                                      discover upcoming concerts around you
                                                            </small>
                                                  </NavbarBrand>
                                        </NavbarHeader>
                              </Col>
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={6}
                              >
                                        <Connect(Filters)
                                                  requestEvents={[Function]}
                                        />
                              </Col>
                    </Row>
          </div>
          <Row
                    bsClass="row"
                    className="map-container"
                    componentClass="div"
          >
                    <Connect(Map) />
          </Row>
          <div
                    className="playlist-container"
          >
                    <Connect(Playlist) />
                    <ScrollWrapper
                              className="react-scrollbar-default"
                              speed={53}
                              style={
                                        Object {
                                                  "height": "290",
                                                  "width": "100%",
                                                }
                              }
                    >
                              <Connect(Concerts)
                                        events={undefined}
                                        handleArtistClick={[Function]}
                              />
                    </ScrollWrapper>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="concert-mate-app"
>
          <div
                    className="container"
          >
                    <Row
                              bsClass="row"
                              componentClass="div"
                    >
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={6}
                              >
                                        <NavbarHeader
                                                  style={
                                                            Object {
                                                                      "color": "#CD3F2A",
                                                                      "fontFamily": "futura",
                                                                      "fontSize": 70,
                                                                    }
                                                  }
                                        >
                                                  <NavbarBrand>
                                                            <a
                                                                      href="/"
                                                            >
                                                                      ConcertMate
                                                            </a>
                                                            <small>
                                                                      discover upcoming concerts around you
                                                            </small>
                                                  </NavbarBrand>
                                        </NavbarHeader>
                              </Col>
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={6}
                              >
                                        <Connect(Filters)
                                                  requestEvents={[Function]}
                                        />
                              </Col>
                    </Row>
          </div>
          <Row
                    bsClass="row"
                    className="map-container"
                    componentClass="div"
          >
                    <Connect(Map) />
          </Row>
          <div
                    className="playlist-container"
          >
                    <Connect(Playlist) />
                    <ScrollWrapper
                              className="react-scrollbar-default"
                              speed={53}
                              style={
                                        Object {
                                                  "height": "290",
                                                  "width": "100%",
                                                }
                              }
                    >
                              <Connect(Concerts)
                                        events={undefined}
                                        handleArtistClick={[Function]}
                              />
                    </ScrollWrapper>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
